src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -227,10 +230,7 @@ public void ConcurrentSendReceive()
                 Task.WaitAll(writes);
                 Task.WaitAll(reads);
 
-                for (int i = 0; i < sendData.Length; i++)
-                {
-                    Assert.True(sendData[i] == receiveData[i], $"Different at {i}");
-                }
+                Assert.Equal(sendData.OrderBy(i => i), receiveData.OrderBy(i => i));
```

Comment:
This shouldn't be needed...these I/Os should complete in the order in which they were issued.